Leading space in to_string, #11833

Vutsuak16 opened the pull request:
closes #11833 

Final Commits

Vutsuak16 committed:
issue-11833:modified format.py

Vutsuak16 committed:
issue-11833:modified format.py

Vutsuak16 committed:
introduced .strip()

Vutsuak16 committed:
.travis.yml changed again

Vutsuak16 committed:
modified the travis file and introduced .strip()

Vutsuak16 committed:
trigger travis

Vutsuak16 committed:
trigger travis

Vutsuak16 committed:
test_format.py changed

Vutsuak16 committed:
test_format.py changed

Vutsuak16 committed:
test_format.py changed again

 A test case for Series class also created, nd few optimizations on DataFrameFormatter class

Vutsuak16 committed:
test_format.py changed again

Vutsuak16 commented:
@jreback  @jorisvandenbossche  it has passed the test time !!
Had to link with travis to run them locally
Please have a look at the commit

jreback commented:
needs a note in whatsnew, with an example as this is techincally an API change.

jreback commented on pandas/core/format.py:
```diff
@@ -578,7 +578,11 @@ def to_string(self):
 
-        self.buf.writelines(text)
+        if self.index==True:
+            self.buf.writelines(text)
```
more like:

```
# put a nice comment here
if not self.index:
     text = text.replace(......)
self.buf.writelines(text)
```

jreback commented on pandas/tests/test_format.py:
```diff
@@ -1725,7 +1725,7 @@ def test_to_string_no_index(self):
 
         df_s = df.to_string(index=False)
-        expected = " x  y\n 1  4\n 2  5\n 3  6"
+        expected = "x  y\n1  4\n2  5\n3  6"
```
since we don't appear to have a series tests for this, pls add one

Vutsuak16 committed:
A test case for Series class also created, nd few optimizations on DataFrameFormatter class

Vutsuak16 commented:
@jreback   @jorisvandenbossche  do i have to update the what's new section in the documentation?
I have no experience with Pandas workflow, please guide me here?

TomAugspurger commented:
http://pandas.pydata.org/pandas-docs/stable/contributing.html, specifically http://pandas.pydata.org/pandas-docs/stable/contributing.html#documenting-your-code for the whatsnew.

Vutsuak16 committed:
whatsnew added

Vutsuak16 committed:
whatsnew added

Few other optimizations in the doc and altered the comments in formaty.py

doc modifications

Vutsuak16 commented:
@TomAugspurger  @jreback 
I did the changes in the documentation
With the recent commit

jreback commented on doc/source/whatsnew/v0.18.0.txt:
```diff
@@ -161,6 +161,46 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
+
+   df=pd.DataFrame({'a':range(5),'b':range(5)})
+   df.to_string(index=False)
+   u' a  b\n 0  0\n 1  1\n 2  2\n 3  3\n 4  4'
```
you need to mark this ``code-block:: python``, then paste in actual ipython output

jreback commented on doc/source/whatsnew/v0.18.0.txt:
```diff
@@ -161,6 +161,46 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
+
+   df=pd.DataFrame({'a':range(5),'b':range(5)})
+   df.to_string(index=False)
+   u'a  b\n0  0\n1  1\n2  2\n3  3\n4  4'
```
you don't need the output line, ``ipython`` will make it here

jreback commented on doc/source/whatsnew/v0.18.0.txt:
```diff
@@ -161,6 +161,46 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
+
+.. ipython:: python
+
+   df=pd.DataFrame({'a':range(5),'b':range(5)})
```
put the first line in its own ``ipython:: python`` block

jreback commented on doc/source/whatsnew/v0.18.0.txt:
```diff
@@ -161,6 +161,46 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
+   u'a  b\n0  0\n1  1\n2  2\n3  3\n4  4'
+
+
+Previous Behaviour for Series Objects
```
skip doing this for Series (just say it applies both to Series & DataFrame above)

jreback commented on doc/source/whatsnew/v0.18.0.txt:
```diff
@@ -161,6 +161,46 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
 
+
+Change in to_string(index=False) method of DataFrameFormatter and SeriesFormatter class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
the underline must match the length of the stop. be a bit more concise on the title.

jreback commented on doc/source/whatsnew/v0.18.0.txt:
```diff
@@ -161,6 +161,46 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
+Change in to_string(index=False) method of DataFrameFormatter and SeriesFormatter class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- the "leading whitespaces" have been removed from the output of "to_string(index=False)" method (:issue: '11833')
```
no space after ``:issue:``

jreback commented on pandas/core/format.py:
```diff
@@ -578,6 +588,18 @@ def to_string(self):
+        ''' this part checks for the argument index passed.
+         If it is true then  returns the text with index mentioned else it returns without the index,
+          the leading whitespaces ,in place of index are also omitted when the index is False
+          ex:
```
only need a single line comment (what I meant)

jreback commented on pandas/core/format.py:
```diff
@@ -242,10 +242,20 @@ def to_string(self):
             fmt_values.insert(row_num + n_header_rows, dot_str)
             fmt_index.insert(row_num + 1, '')
 
+        ''' this part checks for the argument index passed.
```
only need a single line comment

jreback commented:
- some comments
- remove the html-styling file from the commits.
- squash into a single commit.

Vutsuak16 committed:
Few other optimizations in the doc and altered the comments in formaty.py

Vutsuak16 commented:
@jreback  is there anything else that needs attention , for the issue #11833 , I did the docs changes as directed by you

jreback commented on pandas/core/format.py:
```diff
@@ -578,6 +580,11 @@ def to_string(self):
                 text = self.adj.adjoin(1, *strcols)
 
+
+        '''condition check for index '''
```
here too

jreback commented on pandas/core/format.py:
```diff
@@ -242,10 +242,12 @@ def to_string(self):
             fmt_values.insert(row_num + n_header_rows, dot_str)
             fmt_index.insert(row_num + 1, '')
 
+        ''' condition check for index '''
```
take out these comments

jreback commented on doc/source/whatsnew/v0.18.0.txt:
```diff
@@ -161,6 +161,30 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
+Changed to_string(index=False) method of DataFrame and Series formatter class.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- the "leading whitespaces" have been removed from the output of "to_string(index=False)" method (:issue:'11833')
```
on 2nd thought, let's just use this single line 168 instead, pls remove the section. also put double backticks around ``to_string(index=False)``

jreback commented:
need to remove the html-styling ipynb that is included
pls squash to a single commit

Vutsuak16 committed:
commit :

Vutsuak16 commented:
@jreback  I did the needfull ,ignore the last commit :)

jreback commented on doc/source/whatsnew/v0.18.0.txt:
```diff
@@ -161,6 +161,35 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
 .. _whatsnew_0180.api_breaking:
 
+
+<<<<<<< HEAD
```
you have unresolved commits here

jreback commented:
you need to squash: http://pandas.pydata.org/pandas-docs/stable/contributing.html#combining-commits
